<table><tr><td><a href="https://mybinder.org/v2/gh/markcrowe-com/statistics/master?filepath=statistics-lab-test-2021/statistics-lab-test-2021-solution.ipynb" target="_parent"><img src="https://mybinder.org/badge_logo.svg" alt="Open In Binder"/></a></td><td>online editors</td><td><a href="https://colab.research.google.com/github/markcrowe-com/statistics/blob/master/statistics-lab-test-2021/statistics-lab-test-2021-solution.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a></td></tr></table>

# Statistics for Data Analysis

In [3]:
is_remote = False
filename = "statistics-lab-test-2021-solution.xlsx"
remote_filepath = (
    "https://github.com/marcocrowe/statistics/blob/master/statistics-lab-test-2021/"
    + filename
    + "?raw=true"
)
filepath = remote_filepath if is_remote else filename

In [4]:
# local
filepath

'statistics-lab-test-2021-solution.xlsx'

In [3]:
import pandas

In [ ]:

excelWorkbook = pandas.ExcelFile(filepath)

Given the sample space for the experiment of throwing 2 dice determine the following:

| Dice 1 | Dice 2 | Dice 1 | Dice 2 | Dice 1 | Dice 2 | Dice 1 | Dice 2 | Dice 1 | Dice 2 | Dice 1 | Dice 2 |
|--------|--------|--------|--------|--------|--------|--------|--------|--------|--------|--------|--------|
| 1      | 1      | 1      | 2      | 1      | 3      | 1      | 4      | 1      | 5      | 1      | 6      |
| 2      | 1      | 2      | 2      | 2      | 3      | 2      | 4      | 2      | 5      | 2      | 6      |
| 3      | 1      | 3      | 2      | 3      | 3      | 3      | 4      | 3      | 5      | 3      | 6      |
| 4      | 1      | 4      | 2      | 4      | 3      | 4      | 4      | 4      | 5      | 4      | 6      |
| 5      | 1      | 5      | 2      | 5      | 3      | 5      | 4      | 5      | 5      | 5      | 6      |
| 6      | 1      | 6      | 2      | 6      | 3      | 6      | 4      | 6      | 6      | 5      | 6      |

## Q4.1.1. The relative frequency of Dice 2 = 4 , display your answer to 4 decimal places.



Each dice has 6 sides, so the probability of rolling a 6 is $\frac{1}{6}$. The probability of rolling a 6 on two dice is $\frac{1}{6} * \frac{1}{6} = \frac{1}{36}$.  
Dice 2 is 4 Dice 1 and Dice 2 are (1, 4) (2, 4) (3, 4) (4, 4) (5, 4) (6, 4).  
Sum is of possible outcome is 6  
Total possible outcomes is 36  
$\frac{6}{36} = \frac{1}{6} = 0.1667$

## Q4.1.2 The joint probability of Dice #1 = 4 and Dice #2 = 3. display your answer to 4 decimal places.

Each dice has a total of 6 sides, The total number of possible outcomes is 6 for one dice.  S= 6  

$ P(Dice1=4 \cap Dice2=3) $  
$ P(Dice1=4) = \frac{1}{6} $  
$P(\text{Dice2}=3) = \frac{1}{6}$  
$ P(Dice1=4 \cap Dice2=3) = P(Dice1=4) * P(Dice2=3) = \frac{1}{6} * \frac{1}{6} = \frac{1}{36} = 0.0278$

## Q4.1.3 The probability of Dice 1 been an even number given that Dice 2 = 6

The probability of Dice 1 been an even number given that Dice 2 = 6. display your answer to 4 decimal places.  
Count of Even number outcome on a dice is $\frac{3}{6} = \frac{1}{2} = 0.5$

$P(Dice1=Even | Dice2=6) = \frac{P(Dice1=Even \cap Dice2=6)}{P(Dice2=6)}$  
$P(Dice1=Even | Dice2=6) = \frac{P(Dice1=Even \cap Dice2=6)}{P(Dice2=6)} = \frac{3}{36} / \frac{6}{36} = \frac{3}{36} * \frac{36}{6} = \frac{3}{6} = 0.5$

### Q4.2 Bayes Theorem is a mathematical formula for calculating conditional probabilities. Define Bayes Theorem, use both the definition and formula to clarify your answer.



Bayes' theorem describes the probability of an event, based on prior knowledge of conditions that might be related to the event.

*For example:* if I throw 2 dice and I know that one of the dice is 4, we can use Bayes' theorem to calculate the probability of the other dice being 3.  The total count of possible outcomes is 36. Knowing the value of dice 2 is 4, the possible outcomes are (1, 4) (2, 4) (3, 4) (4, 4) (5, 4) (6, 4). The total count of possible outcomes is 6. The probability of dice 1 being 3 given that dice 2 is 4 is $\frac{1}{6}$.
$P(A|B) = \frac{P(B|A) * P(A)}{P(B)}$


Relative Frequency is the ratio (fraction or proportion) of the number of times an event occurs to the total number of trials or observations.

There axioms of probability theory are:

1. The probability of an event is a number between 0 and 1.
2. The probability of the sample space is 1.
3. Complementary probability: The probability of an event not occurring is 1 minus the probability of the event occurring. $P('A) = 1 - P(A)$
4.  The probability of the union of two mutually exclusive events is the sum of the probabilities of the individual events. The probability of the union of two events is the sum of the probabilities of the individual events minus the probability of the intersection of the two events. $P(A \cup B) = P(A) + P(B) - P(A \cap B)$
5.  Conditional probability: The probability of event A given that event B has occurred is the probability of the intersection of A and B divided by the probability of B. $P(A|B) = \frac{P(A \cap B)}{P(B)}$

## Q4.3 - training dataset
The following is a training dataset of commitment levels of work for a module and a corresponding variable ‘Pass’ (suggesting the possibility of a student passing the module).

| Commitment | Pass |
|------------|------|
| High       | No   |
| Medium     | Yes  |
| Low        | Yes  |
| High       | Yes  |
| High       | Yes  |
| Medium     | Yes  |
| Low        | No   |
| Low        | No   |
| High       | Yes  |
| Low        | Yes  |
| High       | No   |
| Medium     | Yes  |
| Medium     | Yes  |
| Low        | No   |

Using Bayes Theorem determine the probability of a person Passing the module
given that their commitment level was high.
P(Pass =Yes | Commitment = High)

$P(Pass = Yes | Commitment = High) = \frac{P(Commitment = High) \cap P(Pass = Yes)}{P(Pass = Yes)}$




The total count of yes from the table is 9.
The count of no from the table is 5.

The count of high from the table is 5.

$P(Pass = Yes | Commitment = High) = \dfrac{P(Commitment = High) \cap P(Pass = Yes)}{P(Pass = Yes)}$  

$P(Pass = Yes) = \dfrac{9}{14} = 0.6429 $  
$P(Commitment = High) = \dfrac{5}{14} = 0.3571$  
$P(Commitment = High) \cap P(Pass = Yes) = \dfrac{5}{14} * \dfrac{9}{14} = \dfrac{45}{196} = 0.2296$  
$P(\text{Pass = Yes | Commitment = High}) = \dfrac{\frac{45}{196}}{\frac{9}{14}} = \dfrac{45}{196} / \dfrac{9}{14} = \dfrac{45}{196} * \dfrac{14}{9} = \dfrac{45}{126} = 0.3571$



## Other stuff



### Answer: Diagram

[![](https://mermaid.ink/img/eyJjb2RlIjoiZ3JhcGggVEI7XG5cdHN0YXJ0KCApXG5cdHN0YXJ0LS0-IHwwLjIwfCBYKChcIlAoWClcIikpXG5cdHN0YXJ0LS0-IHwwLjgwfCBaKChcIlAoWilcIikpXG5cdFgtLT4gfFwiUChCfFgpPTAuMDRcInwgWEIoKFwiUChYJmNhcDtCKVwiKSlcblx0WC0tPiB8XCJQKELMhXxYKT0wLjk2XCJ8IFhiKChcIlAoWCZjYXA7QsyFKVwiKSlcblx0Wi0tPiB8XCJQKEJ8Wik9MC4wMlwifCBaQigoXCJQKFomY2FwO0IpXCIpKVxuXHRaLS0-IHxcIlAoQsyFfFopPTAuOThcInwgWmIoKFwiUChaJmNhcDtCzIUpXCIpKVxuXHRYQi0tPlhCQW5zd2VyKDAuMiAqIDAuMDQgPSAwLjAwOClcblx0WGItLT5YYkFuc3dlcigwLjIgKiAwLjk2ID0gMC4xOTIpXG5cdFpCLS0-WkJBbnN3ZXIoMC44ICogMC4wMiA9IDAuMDE2KVxuXHRaYi0tPlpiQW5zd2VyKDAuOCAqIDAuOTggPSAwLjc4NClcblx0WEJBbnN3ZXItLT5CKChcIlAoQilcIikpXG5cdFpCQW5zd2VyLS0-QlxuXHRCLS0-QkFuc3dlcihcIjAuMDA4ICsgMC4wMTYgPSAwLjAyNFwiKSIsIm1lcm1haWQiOnsidGhlbWUiOiJkZWZhdWx0In0sInVwZGF0ZUVkaXRvciI6ZmFsc2UsImF1dG9TeW5jIjp0cnVlLCJ1cGRhdGVEaWFncmFtIjpmYWxzZX0)](https://mermaid-js.github.io/mermaid-live-editor/edit/#eyJjb2RlIjoiZ3JhcGggVEI7XG5cdHN0YXJ0KCApXG5cdHN0YXJ0LS0-IHwwLjIwfCBYKChcIlAoWClcIikpXG5cdHN0YXJ0LS0-IHwwLjgwfCBaKChcIlAoWilcIikpXG5cdFgtLT4gfFwiUChCfFgpPTAuMDRcInwgWEIoKFwiUChYJmNhcDtCKVwiKSlcblx0WC0tPiB8XCJQKELMhXxYKT0wLjk2XCJ8IFhiKChcIlAoWCZjYXA7QsyFKVwiKSlcblx0Wi0tPiB8XCJQKEJ8Wik9MC4wMlwifCBaQigoXCJQKFomY2FwO0IpXCIpKVxuXHRaLS0-IHxcIlAoQsyFfFopPTAuOThcInwgWmIoKFwiUChaJmNhcDtCzIUpXCIpKVxuXHRYQi0tPlhCQW5zd2VyKDAuMiAqIDAuMDQgPSAwLjAwOClcblx0WGItLT5YYkFuc3dlcigwLjIgKiAwLjk2ID0gMC4xOTIpXG5cdFpCLS0-WkJBbnN3ZXIoMC44ICogMC4wMiA9IDAuMDE2KVxuXHRaYi0tPlpiQW5zd2VyKDAuOCAqIDAuOTggPSAwLjc4NClcblx0WEJBbnN3ZXItLT5CKChcIlAoQilcIikpXG5cdFpCQW5zd2VyLS0-QlxuXHRCLS0-QkFuc3dlcihcIjAuMDA4ICsgMC4wMTYgPSAwLjAyNFwiKSIsIm1lcm1haWQiOiJ7XG4gIFwidGhlbWVcIjogXCJkZWZhdWx0XCJcbn0iLCJ1cGRhdGVFZGl0b3IiOmZhbHNlLCJhdXRvU3luYyI6dHJ1ZSwidXBkYXRlRGlhZ3JhbSI6ZmFsc2V9)

$P(X|B) = \dfrac {P(X \cap B)}{P(B)}$  

$P(X|B) = \dfrac {0.008}{0.024} = \dfrac{1}{3}$

### Answer: Contingency Table

| | $$ B \text{ (broken)} $$ | $$ \overline{B} \text{ (not broken)} $$  | |
|-|-|-|-|
| $$ A \text{ (Dishwasher X)} $$ | $$ A \cap B $$  $$ 0.2 * 0.04 = 0.008 $$ | $$ A \cap \overline{B} $$ $$ 0.2 * 0.96 = 0.192 $$ | $$ 0.2 $$ |
| $$ \overline{A} \text{ (Dishwasher Z)} $$ | $$ \overline{A} \cap B $$ $$ 0.8 * 0.02 = 0.016 $$ | $$ \overline{A} \cap \overline{B} $$  $$ 0.8 * 0.98 = 0.784 $$ | $$ 0.8 $$ |
| | $$ 0.024 $$ | $$ 0.976 $$ | $$ 1.00 $$|


$ P(A \cap B) = 0.008 $  
$ P(B) = 0.024 $  
$P(A|B) = \dfrac {P(A \cap B)}{P(B)}$  

$P(A|B) = \dfrac {0.008}{0.024} = \dfrac{1}{3}$

### Answer: Contingency Table Excel

In [5]:
excelWorkbook.parse('Question1')

X  P(X)  P(B|X)  P(B̅|X)  P(X∩B)  P(X∩B̅)   P(B)  P(B̅)    P(X|B)
0   A   0.2    0.04     0.96   0.008    0.192  0.024  0.976  0.333333
1  A̅   0.8    0.02     0.98   0.016    0.784  0.024  0.976  0.666667

$ P(X|B) = 0.333333 $

## Question 2
Vehicles connected by two sub-routes pass through a route. 10 % of the vehicles from sub-route A are vans while B holds 20% of this type of vehicle. Also, 40% of the vehicles that pass through the route come from the sub-route A. If we pick a sample of 3 vehicles, what is the probability to find 2 vans within them?

### ContingencyTable

| | $$ B \text{ (van)} $$ | $$ \overline{B} \text{ (not a van)} $$  | |
|-|-|-|-|
| $$ A \text{ (route A)} $$ | $$ 0.4 * 0.1 = 0.04 $$ | $$ 0.4 * 0.9 = 0.36 $$ | $$ 0.4 $$ |
| $$ \overline{A} \text{ (route B)} $$ | $$ 0.6 * 0.2 = 0.12 $$ | $$ 0.6 * 0.8 = 0.48 $$ | $$ 0.6 $$ |
| | $$ 0.16 $$ | $$ 0.84 $$ | $$ 1.00 $$|

$ P(V) = 0.16 $  

In [6]:
excelWorkbook.parse('Question2Part1')

X  P(X)  P(B|X)  P(B̅|X)  P(X∩B)  P(X∩B̅)  P(B)  P(B̅)
0   A   0.4     0.1      0.9    0.04     0.36  0.16   0.84
1  A̅   0.6     0.2      0.8    0.12     0.48  0.16   0.84

In [1]:
from scipy.stats import binom

In [7]:
excelWorkbook.parse('Question2Part2')

Vans  P(V)  Sample Size  BINOM.DIST  BINOM.DIST(Mass)  BINOM.DIST(Mass)2
0     0  0.16            3    0.592704          0.592704           0.592704
1     1  0.16            3    0.931392          0.338688           0.338688
2     2  0.16            3    0.995904          0.064512           0.064512
3     3  0.16            3    1.000000          0.004096           0.004096

Answer P(2Vans in Sample of 3) = 0.064512  

## Question 3
A machine has certain number of failures that cannot exceed 3 failures a day. The following table represents the probability distribution for this variable.

Number of failures|0|1|2|3
-|-|-|-|-
Probabilities|?|?|?|0.2

Complete the table knowing that the probability of having less than 2 failures is 50% and the average of failures in 5 working days is 6.5 and calculate the variance.

### Answer
In the question we learn the **"probability of having less than 2 failures is 50%"**  
$ \therefore P(0) + P(1) = 0.5 $   
$ \sum P(X) = P(0) + P(1) + P(2) + P(3) = 1 $  
$ P(0) + P(1) + P(2) + 0.2 = 1 $  
$ 0.5 + P(2) + 0.2 = 1 $  
$ P(2) = 0.3 $ 

In the question we learn the **"the average of failures in 5 working days is 6.5"**  
$ \therefore \overline{x} = \frac{6.5}{5} = 1.3 $  

$ 0*P(0) + 1*P(1) + 2*P(2) + 3*P(3) = \overline{x} $  
$ 0 + 1*P(1) + 2*0.3 + 3*0.2 = 1.3 $  
$ 0 + 1*P(1) + 2*0.3 + 3*0.2 = 1.3 $  
$ P(1) + 0.6 + 0.6 = 1.3 $  
$ P(1) + 1.2 = 1.3 $  
$ P(1) = 0.1 $  

$ P(0) + P(1) = 0.5 $   
$ P(0) + 0.1 = 0.5 $  
$ P(0) = 0.4 $  

Number of failures|0|1|2|3
-|-|-|-|-
Probabilities|0.4|0.1|0.3|0.2

In [8]:
excelWorkbook.parse('Question3')

X  P(X)  X * P(X)   x̄  X-∑(x̄)  (X -∑(x̄))^2  P(X) * (X - ∑(x̄))^2   S^2  \
0  0   0.4       0.0  1.3     -1.3          1.69                 0.676  1.41   
1  1   0.1       0.1  1.3     -0.3          0.09                 0.009  1.41   
2  2   0.3       0.6  1.3      0.7          0.49                 0.147  1.41   
3  3   0.2       0.6  1.3      1.7          2.89                 0.578  1.41   

          S  
0  1.187434  
1  1.187434  
2  1.187434  
3  1.187434

The Variance, $S^2 = 1.41 $

## Question 4
The following table display the results after flipping 6 coins 64 times. What is the standard deviation?

Number of Tails|0|1|2|3|4|5|6
-|-|-|-|-|-|-|-
Frequency|1|6|15|20|15|6|1

### Answer 

In [9]:
excelWorkbook.parse('Question4')

X  Frequency   n      P(X)  ∑P(X)  X * P(X)  E(X)  X-E(X)  (X - E(X))^2  \
0  0          1  64  0.015625      1   0.00000     3      -3             9   
1  1          6  64  0.093750      1   0.09375     3      -2             4   
2  2         15  64  0.234375      1   0.46875     3      -1             1   
3  3         20  64  0.312500      1   0.93750     3       0             0   
4  4         15  64  0.234375      1   0.93750     3       1             1   
5  5          6  64  0.093750      1   0.46875     3       2             4   
6  6          1  64  0.015625      1   0.09375     3       3             9   

   P(X) * (X - E(X))^2  S^2         S  
0             0.140625  1.5  1.224745  
1             0.375000  1.5  1.224745  
2             0.234375  1.5  1.224745  
3             0.000000  1.5  1.224745  
4             0.234375  1.5  1.224745  
5             0.375000  1.5  1.224745  
6             0.140625  1.5  1.224745

Standard deviation, $ S = 1.224745 $

### Answer 

In [10]:
excelWorkbook.parse('Question4Alt1')

X  Frequency   n  x̅  X-x̅  (X - x̅)^2 * Freq  ∑((X - x̅)^2 * Freq)  S^2  \
0  0          1  64   3    -3                  9                    96  1.5   
1  1          6  64   3    -2                 24                    96  1.5   
2  2         15  64   3    -1                 15                    96  1.5   
3  3         20  64   3     0                  0                    96  1.5   
4  4         15  64   3     1                 15                    96  1.5   
5  5          6  64   3     2                 24                    96  1.5   
6  6          1  64   3     3                  9                    96  1.5   

          S  
0  1.224745  
1  1.224745  
2  1.224745  
3  1.224745  
4  1.224745  
5  1.224745  
6  1.224745

Standard deviation, $ S = 1.224745 $

### Answer Population

In [11]:
excelWorkbook.parse('Question4Alt2')

0 Tails  1  Tails  2  Tails  3  Tails  4  Tails  5  Tails  6  Tails  \
0       0.0       1.0       2.0         3       4.0       5.0       6.0   
1       NaN       1.0       2.0         3       4.0       5.0       NaN   
2       NaN       1.0       2.0         3       4.0       5.0       NaN   
3       NaN       1.0       2.0         3       4.0       5.0       NaN   
4       NaN       1.0       2.0         3       4.0       5.0       NaN   
5       NaN       1.0       2.0         3       4.0       5.0       NaN   
6       NaN       NaN       2.0         3       4.0       NaN       NaN   
7       NaN       NaN       2.0         3       4.0       NaN       NaN   
8       NaN       NaN       2.0         3       4.0       NaN       NaN   
9       NaN       NaN       2.0         3       4.0       NaN       NaN   
10      NaN       NaN       2.0         3       4.0       NaN       NaN   
11      NaN       NaN       2.0         3       4.0       NaN       NaN   
12      NaN       NaN       2.0         3       4.0       NaN       NaN   
13      NaN       NaN       2.0         3       4.0       NaN       NaN   
14      NaN       NaN       2.0         3       4.0       NaN       NaN   
15      NaN       NaN       NaN         3       NaN       NaN       NaN   
16      NaN       NaN       NaN         3       NaN       NaN       NaN   
17      NaN       NaN       NaN         3       NaN       NaN       NaN   
18      NaN       NaN       NaN         3       NaN       NaN       NaN   
19      NaN       NaN       NaN         3       NaN       NaN       NaN   
20      1.0       6.0      15.0        20      15.0       6.0       1.0   

    Average  Population Standard deviation  
0       3.0                       1.224745  
1       3.0                       1.224745  
2       3.0                       1.224745  
3       3.0                       1.224745  
4       3.0                       1.224745  
5       3.0                       1.224745  
6       3.0                       1.224745  
7       3.0                       1.224745  
8       3.0                       1.224745  
9       3.0                       1.224745  
10      3.0                       1.224745  
11      3.0                       1.224745  
12      3.0                       1.224745  
13      3.0                       1.224745  
14      3.0                       1.224745  
15      3.0                       1.224745  
16      3.0                       1.224745  
17      3.0                       1.224745  
18      3.0                       1.224745  
19      3.0                       1.224745  
20      NaN                            NaN

Standard deviation, $ S = 1.224745 $

## Question 5
The median is the second quartile of the sample, True or False?  
***True***
## Question 6
Mutual exclusive events are those that are not related to each other, it means that the occurrence of one of them does not affect the occurrence of the other one, True or False?  
***False***
## Question 7
The mode and the median have always the same result, True or False?  
***False***
## Question 8
Binomial distribution is useful when you want to know the probability of success of certain event without taking into consideration the order in which the element appears, True or False?  
***True***

In [12]:
# Close Connection to the open excelWorkbook
excelWorkbook.close()

Copyright (c) 2021 Mark Crowe (<https://github.com/markcrowe-com>). All rights reserved.  